### DISCUSSION TITLE
Joining a table to a view in select()

### DISCUSSION THREAD

**Discussion by jnorris441** [supabase/supabase#5919] [Questions] [answered]
Hi I have a query that need to include a few tables and a view. 

I can include the tables because of the foreign key relationships, but the view has no foreign key. 

How do you manually join the view in the query? If not is that a limitation of PostgREST?

**Comment by steve-chavez** [comment, marked as answer]
> How do you manually join the view in the query?

Try it in the same way as tables, PostgREST infers view's foreign keys. See https://postgrest.org/en/stable/api.html#embedding-views

**Reply by jnorris441** [reply]
TABLE **projects** 
VIEW **projects_stats** 

**projects_stats** has a project_id field but it's not enough for PostgREST to figure it out 🤷‍♂️

**Reply by jnorris441** [reply]
However that is useful to know and I will keep it in mind.

**Reply by steve-chavez** [reply]
> projects_stats has a project_id field but it's not enough for PostgREST to figure it out man_shrugging

Hm, what error do you get?

**Reply by jnorris441** [reply]
Are you saying it should be able to infer that `projects_stats.project_id` is a reference to `projects.id`? 

```
{
    "message": "Could not find a relationship between 'projects' and 'projects_stats' in the schema cache",
    "hint": "Verify that 'projects' and 'projects_stats' exist in the schema 'public' and that there is a foreign key relationship between them. If a new relationship was created, try reloading the schema cache."
}
```

**Reply by steve-chavez** [reply]
Yes, can you share the tables/views definitions? That way I can see what could be wrong.

**Reply by jnorris441** [reply]
I have since replaced this view with columns updated by a trigger, but I am interested in why it doesn't work.

```
CREATE TABLE IF NOT EXISTS public.projects
(
    created_date timestamp with time zone DEFAULT now(),
    name character varying COLLATE pg_catalog."default",
    profile_image character varying COLLATE pg_catalog."default",
    cover_image character varying COLLATE pg_catalog."default",
    website_url character varying COLLATE pg_catalog."default",
    id uuid NOT NULL DEFAULT uuid_generate_v4(),
    contract_address character varying COLLATE pg_catalog."default",
    founded_date timestamp with time zone,
    discord_url character varying COLLATE pg_catalog."default",
    twitter_url character varying COLLATE pg_catalog."default",
    marketplace_url character varying COLLATE pg_catalog."default",
    instagram_url character varying COLLATE pg_catalog."default",
    blockchain_id uuid,
    account_id uuid,
    contract_url character varying COLLATE pg_catalog."default",
    description text COLLATE pg_catalog."default",
    reporttypes_negative_count bigint DEFAULT '0'::bigint,
    reporttypes_positive_count bigint DEFAULT '0'::bigint,
    reports_positive_count bigint DEFAULT '0'::bigint,
    reports_negative_count bigint DEFAULT '0'::bigint,
    CONSTRAINT project_pkey PRIMARY KEY (id),
    CONSTRAINT projects_blockchain_id_fkey FOREIGN KEY (blockchain_id)
        REFERENCES public.blockchains (id) MATCH SIMPLE
        ON UPDATE NO ACTION
        ON DELETE NO ACTION
)
```


```
CREATE view projects_stats
AS SELECT id as project_id, 
(SELECT COUNT(*) from reports where active=true AND project_id = projects.id AND positive = true) as reports_positive_count,
(SELECT COUNT(*) FROM (SELECT COUNT(*) from reports where active=true AND positive = true AND project_id = projects.id GROUP BY reporttype_id) as reportcount) as reporttypes_positive_count,
(SELECT COUNT(*) from reports where active=true AND project_id = projects.id AND positive = false) as reports_negative_count,
(SELECT COUNT(*) FROM (SELECT COUNT(*) from reports where active=true AND positive = false AND project_id = projects.id GROUP BY reporttype_id) as reportcount) as reporttypes_negative_count
FROM projects
```



The projects table was named project originally. I wonder if it has to do with the project table's pkey constraint having the name project_pkey and not projects_pkey? 

https://github.com/supabase/supabase/issues/5549

**Reply by steve-chavez** [reply]
> The projects table was named project originally. I wonder if it has to do with the project table's pkey constraint having the name project_pkey and not projects_pkey?

@jnorris441 No, that shouldn't affect the detection.

The thing is that `projects_stats` is "inheriting" the foreign keys of `projects`(because it has `FROM projects` in its definition) - that means it has a relationship to `blockchains` and what you can do is join `projects_stats` with `blockchains`, this would only work if you had `blockchain_id` exposed in your view though.

If you'd like joining `projects` with `projects_stats` then `projects` would need a self-reference foreign key or you could also change the view definition to come `FROM reports` and if `reports` has a foreign key to `projects` then the joining should work.

**Reply by jnorris441** [reply]
@steve-chavez that makes sense, thank you
